ddtrace/contrib/django/patch.py

```diff
@@ -439,116 +431,96 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
     if request is None:
         return func(*args, **kwargs)
 
-    trace_utils.activate_distributed_headers(pin.tracer, int_config=config.django, request_headers=request.META)
     request_headers = utils._get_request_headers(request)
 
     with core.context_with_data(
         "django.traced_get_response",
         remote_addr=request.META.get("REMOTE_ADDR"),
         headers=request_headers,
         headers_case_sensitive=django.VERSION < (2, 2),
-    ):
-        with pin.tracer.trace(
-            schematize_url_operation("django.request", protocol="http", direction=SpanDirection.INBOUND),
-            resource=utils.REQUEST_DEFAULT_RESOURCE,
-            service=trace_utils.int_service(pin, config.django),
-            span_type=SpanTypes.WEB,
-        ) as span:
-            core.dispatch(
-                "django.traced_get_response.pre",
-                functools.partial(_block_request_callable, request, request_headers, span),
-            )
-            span.set_tag_str(COMPONENT, config.django.integration_name)
+        span_name=schematize_url_operation("django.request", protocol="http", direction=SpanDirection.INBOUND),
+        resource=utils.REQUEST_DEFAULT_RESOURCE,
+        service=trace_utils.int_service(pin, config.django),
+        span_type=SpanTypes.WEB,
+        tags={COMPONENT: config.django.integration_name, SPAN_KIND: SpanKind.SERVER},
+        distributed_headers_config=config.django,
+        distributed_headers=request_headers,
+        pin=pin,
+    ) as ctx, ctx.get_item("call"):
+        core.dispatch(
+            "django.traced_get_response.pre",
+            functools.partial(_block_request_callable, request, request_headers, ctx),
+            ctx,
+            request,
+            utils._before_request_tags,
+        )
 
-            # set span.kind to the type of request being performed
-            span.set_tag_str(SPAN_KIND, SpanKind.SERVER)
+        response = None
+
+        def blocked_response():
```

Comment:
Why not have this function defined at top level, instead of nested here?

Reply:
That's a fair question. This pull request actually doesn't change where this function is defined, but if I were going to change that I'd tend to share your instinct that top-level is preferable.